website/static/js/fangorn.js

```diff
@@ -737,68 +843,100 @@ function _fangornUploadMethod(item) {
  * @returns {Array} Returns an array of mithril template objects using m()
  * @private
  */
-function _fangornActionColumn (item, col) {
-    var self = this,
+function _fangornDefineToolbar (item) {
```

Comment:
This API is rather opaque; what hapens when you assign `buttons` to `item`? Also, defining buttons as objects with `name`, `template`, etc. seems like reinventing the wheel. Mithril already allows us to group display logic with controllers and views; why not take advantage of that?
I think this API can be vastly improved by allowing the toolbar to be a mithril module/component.

```js
// A fangorn-styled button; addons can reuse this
var Button = {
    controller: function(args) {
        var noop = function() {};
        this.onclick = args.onClick || noop;
    }
    view: function(ctrl, args, children) {
        var extraCSS = args.className || '';
        var tooltipText = args.tooltip || '';
        var iconCSS = args.icon;
        return m('div', {
            className: 'fangorn-toolbar-icon' + extraCSS,
            onclick: ctrl.onclick,
            'data-toggle': tooltipText ? 'tooltip' : '',
            'title':  tooltipText}, [
                m('i', {className: iconCSS}),
                m('span.hidden-sx', children)
            ]);
    }
}

var FangornToolbar = {
    view: function(ctrl, args) {
        var buttons = [
            m.component(Button, {
                onclick: function(evt) {_uploadEvent.call(tb, evt, item)}
                tooltip: 'Select files to upload',
            }, 'Upload')
            m.component(Button, {
                // Treebeard instance can be passed in as an argument
                onclick: function(evt) {args.tb.options.iconState.mode = 'createFolder';}
                tooltip: 'Create a new folder'
            }, 'Create Folder')
        ];  
        // ... logic to add more buttons
        if (item.data.path) { buttons.push(...) }
        return buttons;
    }
}

//...
var tbOptions =  {
    toolbarComponent: FangornToolbar
}
new Fangorn(tbOptions)
```